developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,133 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+Why UUID Needed
+***************
+
+To develope a new audio signal processing component, we traditionally
+implement the component driver with a new unique component type
+introduced, the painpoints with this method including how to keep ABI
+aligned and backward compatible for different version combination of
+topology/driver/firmware for each new added components, there could be
+component type collision if versions are not aligned.
+
+UUIDs (Universally Unique Identifiers) provide a more scalable and
+collision-free way of component identification. UUIDs are currently
```

Comment:
currently? this is still work in progress?

Reply:
I assumed this will get merged after the implementation is merged, so saying "currently" here.

Reply:
OK, let me remove "currently", thanks.